### PR TITLE
revert #1873

### PR DESCRIPTION
**Pull request by tyler-french**
We need access to `rule.Name()` as well as `rule.Kind()` in mapped resolver.

### CODE REVIEW

**Comment by fmeum** [comment]
@jbedard We need to roll this change back as it breaks the public resolver API, which has a legitimate need to access other rule fields. Do you see a way to reland the desired behavior without touching public API?

**Review by fmeum — APPROVED** [review]

File: resolve/index.go [hunk 1]
```diff
@@ -110,6 +110,8 @@ type RuleIndex struct {
 
 // ruleRecord contains information about a rule relevant to import indexing.
 type ruleRecord struct {
+	rule  *rule.Rule
```
Comment (tyler-french) [inline resolve/index.go:113]: This is still serializable since https://pkg.go.dev/encoding/json#Marshal only acts on exported fields of a structure.

Comment (jbedard) [reply resolve/index.go:113]: The problem will be unmarshalling though, won't it?

**Comment by jbedard** [comment]
I'll look at it again if we must revert it. 

Is this actually a publicly documented API though?

**Comment by tyler-french** [comment]
> I'll look at it again if we must revert it.
> 
> Is this actually a publicly documented API though?

https://pkg.go.dev/github.com/bazelbuild/bazel-gazelle@v0.38.0/resolve#NewRuleIndex

**Pull request merged**